lib/backend/model/statusreports.go

```diff
@@ -24,13 +24,14 @@ import (
 )
 
 var (
-	matchActiveStatusReport = regexp.MustCompile("^/?calico/felix/v1/host/([^/]+)/status$")
-	matchLastStatusReport   = regexp.MustCompile("^/?calico/felix/v1/host/([^/]+)/last_reported_status")
+	matchActiveStatusReport = regexp.MustCompile("^/?calico/felix/v2/([^/]+)/host/([^/]+)/status$")
+	matchLastStatusReport   = regexp.MustCompile("^/?calico/felix/v2/([^/]+)/host/([^/]+)/last_reported_status")
 	typeStatusReport        = reflect.TypeOf(StatusReport{})
 )
 
 type ActiveStatusReportKey struct {
-	Hostname string `json:"-" validate:"required,hostname"`
+	Hostname     string `json:"-" validate:"required,hostname"`
+	RegionString string
```

Comment:
Missing the json/validate tags?

Reply:
We should probably validate that it doesn't have a `/` in it if nothing else.

Reply:
Just a thought: could you make RegionString into a typedef?  I.e. `type RegionString string`  Then the user would have to go through the RegionString function.

Reply:
Yes, that's a good idea, and I could make the RegionString function do escaping.

Reply:
Actually no, that requires too much change everywhere, and is inconsistent which all the other string-like fields that are just 'string'.  Let's just do the validation.

Reply:
OK, can live with that; should there be a `validate` tag, then?